x11-misc/xscreensaver/xscreensaver-6.05.ebuild

```
@@ -43,21 +40,19 @@ COMMON_DEPEND="
 		x11-libs/gdk-pixbuf-xlib
 		>=x11-libs/gdk-pixbuf-2.42.0:2
 	)
-	gtk? ( x11-libs/gtk+:2 )
+	gtk? ( >=x11-libs/gtk+-2.22.0:3 )
```

we don't need the 2.22 bit with :3

---

But that's what it says in the configure script.
Upstream is checking for 2.22 even though it now depends on gtk-3.

The configure script is kind of a mess and contradictory on some things. I patched it a bit therefore.

---

I think the point that sam tried to make is that there can be never a version that is lower than 2.22 in the :3 slot in Gentoo. So you can just

```suggestion
	gtk? ( x11-libs/gtk+:3 )
```

which expresses the same condition as before, but is shorter and hence easier to read.

---

Yes I understand. When bumping the gnome 43 packages devs sometimes told me I should put in the version that is checked for in the meson build, even though that version is so old, it's rather unlikely if not impossible that someone has a package this old. This was to not confuse the next one that bumps the package. 

But I can change this to gtk:3, I'm fine with that.

---

The reasoning makes sense, but it's a bit odd they include that constraint even when looking for gtk-3 (which has a different pkg-config name...)